### PR TITLE
Document that a CupertinoApp isn't necessary for an iOS app

### LINKED ISSUE #24766: Document that a CupertinoApp isn't necessary for an iOS app
From https://medium.com/asos-techblog/flutter-vs-react-native-for-ios-android-app-development-c41b4e038db9 it wasn't clear that CupertinoApp wasn't necessary to build an iOS app. Document and add an example.

### PR DESCRIPTION
**Pull request by Abhishek01039**
fixes https://github.com/flutter/flutter/issues/24766
 
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/app.dart [hunk 1]
```diff
@@ -37,6 +37,8 @@ import 'theme.dart';
 /// This widget also configures the observer of the top-level [Navigator] (if
 /// any) to perform [Hero] animations.
 ///
+/// [CupertinoApp] isn't necessary for an iOS app.
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/app.dart:40]: This needs more details and a proposed solution. 

```suggestion
/// The [CupertinoApp] widget isn't a required ancestor for other Cupertino 
/// widgets, but many Cupertino widgets could depend on the [CupertinoTheme]
/// widget, which the [CupertinoApp] composes. If you use Material widgets, a
/// [MaterialApp] also creates the needed dependencies for Cupertino widgets.
```

Comment (Abhishek01039) [reply packages/flutter/lib/src/cupertino/app.dart]: Ah, Awesome
You are right. we need more details
let me add 
Thanks

**Review by xster — APPROVED** [review]

**Comment by xster** [comment]
thanks

**Comment by Abhishek01039** [comment]
Thanks @xster

**Pull request merged**